spec/controllers/beta/projects_controller_spec.rb

```diff
@@ -0,0 +1,63 @@
+require 'rails_helper'
+
+describe Beta::ProjectsController do
+  context 'when logged out' do
+    specify do
+      get :show, params: { id: 42 }
+      expect(response).to redirect_to(new_user_session_url)
+    end
+  end
+
+  describe 'when normal user is logged in' do
+    let(:normal_user)    { create :user, :with_team, email: 'normal_user@example.com' }
+    let(:normal_team)    { normal_user.teams.first }
+    let(:normal_project) { create :project, users: [normal_user] }
+
+    before do
+      normal_team.ownerships.create(project: normal_project, is_owner: true)
+      sign_in normal_user
+    end
+
+    context 'when normal user try to access beta version' do
+      it 'should be redirect' do
+        get :show, params: { id: normal_project.id }
+
+        expect(response).not_to be_successful
+      end
+    end
+  end
+
+  context 'when admin is logged in' do
+    let(:user)    { create :user, :with_team_and_is_admin, email: 'user@example.com' }
+    let(:team)    { user.teams.first }
+    let(:project) { create :project, users: [user] }
+    let!(:story)  { create(:story, project: project, requested_by: user) }
+
+    before do
+      team.ownerships.create(project: project, is_owner: true)
+      sign_in user
+    end
+
+    describe 'collection actions' do
+      describe '#show' do
+        describe 'when try get a invalid project' do
```

Comment:
maybe change just to `tries to get an invalid project` since the context already clarifies that this happens _when an admin is logged in_